torchtune/datasets/multimodal/_the_cauldron.py

```diff
@@ -0,0 +1,165 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Any, Dict, Mapping, Optional, Union
+
+from torchtune.data import Message
+from torchtune.datasets._packed import PackedDataset
+from torchtune.datasets._sft import SFTDataset
+from torchtune.modules.transforms import Transform
+
+
+class TheCauldronToMessages(Transform):
```

Comment:
I do worry a bit that we're overfitting to these two datasets in our choice of naming. Like really iiuc llava_instruct means you have a local directory of images and the sample points to filenames and the cauldron means you have actual PILs in each sample. That seems more general than what we're indicating

Reply:
well, we only have one dataset example that fits this mold. so does it make sense to make it general at this point?

Reply:
Yeah I guess this is kinda inconsistent with my other comment anyways. Let's go with what you proposed there, keep the naming as is and have a separate transform for each of the cauldron and llava-instruct datasets